dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentUrlPatternCalculator.java

```diff
@@ -0,0 +1,86 @@
+package com.dotcms.experiments.business;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.structure.StructureUtil;
+import com.dotmarketing.util.UtilMethods;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Util class to calculate the regex pattern for a given {@link HTMLPageAsset}
+ */
+public enum ExperimentUrlPatternCalculator {
+    INSTANCE;
+
+    public static String REDIRECT_REGEX_TEMPLATE = "(http|https):\\/\\/.*:.*%s(\\?.*)?";
+    public static final String INDEX = "/index";
+
+    /**
+     * Calculate the URL regex pattern for the given {@link HTMLPageAsset},
+     * This pattern is used to match the Event's url page with the {@link HTMLPageAsset} url.
+     *
+     * For example if you create a Experiment using the "/blog/index" page, any of
+     * the following urls will be matched:
+     *
+     * - http://localhost:8080/blog/index
+     * - http://localhost:8080/blog/index?param1=value1&param2=value2
+     * - http://localhost:8080/blog/
+     * - http://localhost:8080/blog
+     * - http://localhost:8080/blog?param1=value1&param2=value2
+     *
+     * If the page use inside the Experiment isuse as Detail Page on any Content Type then is even
+     * more complicated.
+     *
+     * @param htmlPageAsset
+     * @return
+     */
+    public String calculateUrlRegexPattern(final HTMLPageAsset htmlPageAsset) {
+
+        try {
+            String resulRegex = null;
+
+            final List<String> urlMappedPattern = APILocator.getContentTypeAPI(
+                            APILocator.systemUser())
+                    .findUrlMappedPattern(htmlPageAsset.getIdentifier());
+
+            if (UtilMethods.isSet(urlMappedPattern)) {
```

Comment:
this could be encapsulate on Strategy such as StrategyUrlPattern 
test(Map params);
apply(Map params);

Then you create a Collection of them

strategies.stream().filter(strategy -> strategy.test(params)).findFirst().orElseGet(()-> this.defaultStrategy).apply(params);

Reply:
done https://github.com/dotCMS/core/pull/25100/commits/4c90a2b03a71008b46bba98b301e6633eec5a6ef#diff-2069a2eff51802af984f2ef6b926adc6e8d91779ea57354c6f52d975fc857842R47